composer.json

```diff
@@ -37,14 +37,14 @@
         "doctrine/event-manager": "^1.0"
     },
     "require-dev": {
-        "doctrine/coding-standard": "^8.1",
-        "jetbrains/phpstorm-stubs": "^2019.1",
-        "phpstan/phpstan": "^0.12.55",
+        "doctrine/coding-standard": "8.2.0",
+        "jetbrains/phpstorm-stubs": "2019.3",
+        "phpstan/phpstan": "0.12.57",
         "phpstan/phpstan-strict-rules": "^0.12.2",
-        "phpunit/phpunit": "^9.4",
-        "psalm/plugin-phpunit": "^0.13.0",
+        "phpunit/phpunit": "9.5.0",
+        "psalm/plugin-phpunit": "0.13.0",
         "symfony/console": "^2.0.5|^3.0|^4.0|^5.0",
-        "vimeo/psalm": "^4.1"
+        "vimeo/psalm": "4.2.1"
```

Comment:
@nicwortel why did you use 4.3.1 on your PR to 2.12.x? 2.12.x was using 4.1.1, and 3.0.x is using 4.2.1. Using 4.3.1 here results in some errors.

An interesting consequence of removing composer.lock is that now we can see discrepancies in tools version between branches.

Reply:
You are right... my bad :facepalm: 

I could have sworn that I used the exact version as it was locked by the `composer.lock`, but I must have messed up switching between versions at some point.

It seems like I accidentally also moved doctrine/coding-standard from 8.1.0 to 8.2.0.

Reply:
No worries, I'll fix the issues!

Reply:
Ah I'm going to be forced to upgrade to 4.3.2 because we use `strtok()` with one argument in the codebase, and 4.3.1 takes issue with that (https://github.com/vimeo/psalm/issues/4874)